tests/smoke.rs

```diff
@@ -228,3 +242,79 @@ fn is_serde() {
     is_serialize::<backtrace::Backtrace>();
     is_deserialize::<backtrace::Backtrace>();
 }
+
+#[test]
+fn sp_smoke_test() {
+    let mut refs = vec![];
+    recursive_stack_references(&mut refs);
+    return;
+
+    #[inline(never)]
+    fn recursive_stack_references(refs: &mut Vec<usize>) {
+        assert!(refs.len() < 5);
+
+        let x = refs.len();
+        refs.push(&x as *const _ as usize);
+
+        if refs.len() < 5 {
+            recursive_stack_references(refs);
+            eprintln!("exiting: {}", x);
+            return;
+        }
+
+        let mut child_sp = None;
+        let mut child_ref = None;
+        backtrace::trace(make_trace_closure(refs, &mut child_sp, &mut child_ref));
+        eprintln!("exiting: {}", x);
+    }
+
+    // NB: the following `make_*` functions are pulled out of line, rather than
+    // defining their results as inline closures at their call sites, so that
+    // the resulting closures don't have "recursive_stack_references" in their
+    // mangled names.
+
+    fn make_trace_closure<'a>(
+        refs: &'a mut Vec<usize>,
+        child_sp: &'a mut Option<usize>,
+        child_ref: &'a mut Option<usize>,
```

Comment:
I think these two parameters could be local variables moved into the closure, right? I don't think they need to be defined out-of-line and passed as arguments

Reply:
Done.